spec/api/v1/api_crops_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe Api::V1::CropsController do
+
+  before(:each) do
+    @crop = FactoryGirl.create(:tomato)
+    @crop.save
+  end
+
+  describe "GET crops index" do
+    it 'fetches the crops index' do
+      get '/api/v1/crops'
+      response.should be_success
+      response.content_type.should eq("application/json")
+    end
```

Comment:
Good point. I feel like 200 is exactly what we want here, not eg. 206 
"Partial content".

For reference: 
http://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success

On 8/01/2015 7:36 am, Taylor Griffin wrote:
>
> In spec/api/v1/api_crops_controller_spec.rb 
> <https://github.com/Growstuff/growstuff/pull/616#discussion-diff-22613777>:
>
> > @@ -0,0 +1,45 @@
> > +require 'spec_helper'
> > +
> > +describe Api::V1::CropsController do
> > +
> > +  before(:each) do
> > +    @crop = FactoryGirl.create(:tomato)
> > +    @crop.save
> > +  end
> > +
> > +  describe "GET crops index" do
> > +    it 'fetches the crops index' do
> > +      get '/api/v1/crops'
> > +      response.should be_success
> > +      response.content_type.should eq("application/json")
> > +    end
>
> Also, what do you guys think of the be_success matcher? I generally 
> choose the more specific numeric codes and that seems like it might be 
> better when testing an API.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Growstuff/growstuff/pull/616/files#r22613777>.
>

-- 
Alex "Skud" Bayley
skud@growstuff.org
http://growstuff.org/